### PR TITLE
Rename BraintreeCoreSwift and Delete Obj-C Module

### PR DESCRIPTION
**Pull request by jaxdesmarais**
### Summary of changes

- Remove `BraintreeCore` Objective-C module
- Rename `BraintreeCoreSwift` module to `BraintreeCore`
- Update podspec
- Update Package.swift
- Update imports to reference renamed module
- Update Demo

**Note:** This does not fix Cocoapods and SPM imports yet - that will be tackled in a follow up PR

### Checklist

- ~[ ] Added a changelog entry~ these changes are all internal

- @jaxdesmarais

### CODE REVIEW

File: Sources/BraintreeCard/Public/BraintreeCard/BTCardNonce.h [hunk 1]
```diff
@@ -1,9 +1,6 @@
-#if __has_include(<Braintree/BraintreeCard.h>)
-#import <Braintree/BraintreeCore.h>
-#else
-#import <BraintreeCore/BraintreeCore.h>
-#endif
+typedef NS_ENUM(NSInteger, BTCardNetwork);
```
Comment (sshropshire) [inline Sources/BraintreeCard/Public/BraintreeCard/BTCardNonce.h:1]: random q: Was this in a different file initially?

Comment (jaxdesmarais) [reply Sources/BraintreeCard/Public/BraintreeCard/BTCardNonce.h:1]: Yeah, this is part of core and was converted in PR: https://github.com/braintree/braintree_ios/pull/890. We need to forward declare the enum here to use it for the property further down. Previously, we were importing `Braintree.h` here which imported it as part of the `BTEnums.h` file here: https://github.com/braintree/braintree_ios/blob/master/Sources/BraintreeCore/Public/BraintreeCore/BraintreeCore.h#L16

File: IntegrationTests/Braintree-API-Integration-Specs/BTAPIClient_IntegrationTests.m [hunk 2]
```diff
@@ -1,6 +1,5 @@
-#import <BraintreeCore/BraintreeCore.h>
 #import <XCTest/XCTest.h>
-@import BraintreeCoreSwift;
+@import BraintreeCore;
```
Comment (scannillo) [inline IntegrationTests/Braintree-API-Integration-Specs/BTAPIClient_IntegrationTests.m:2]: I remember this pain in the butt formatting we had to do for importing Swift into Obj-C when we first converted PPDataCollector over. We can talk about this pairing later, but IIRC using the `@import` won't work for all package managers.

https://github.com/braintree/braintree_ios/blob/master/DEVELOPMENT.md#importing-a-swift-module-into-obj-c

File: Sources/BraintreeApplePay/BTApplePayCardNonce.m [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#import "BraintreeCoreSwiftImports.h"
+#import "Foundation/Foundation.h"
```
Comment (scannillo) [inline Sources/BraintreeApplePay/BTApplePayCardNonce.m:1]: I see a few new Foundation imports. Curious if these were moved over from `.m`s or why these were added?

File: Sources/BraintreeCore/Public/BraintreeCore/Braintree.h [hunk 4]
```diff
@@ -1,8 +0,0 @@
-#ifndef Braintree_h
-#define Braintree_h
-
-/// This file added to resolve an issue with cross language module support using CocoaPods:
-/// https://github.com/CocoaPods/CocoaPods/issues/9432#issuecomment-570830515
-/// When compiling a Swift module, the `-Swift.h` header references `Braintree.h`, but requires nothing from it.
-
-#endif /* Braintree_h */
```
Comment (scannillo) [inline Sources/BraintreeCore/Public/BraintreeCore/Braintree.h:8]: 🤔 I think we might need to add this file back for CocoaPods. If we add it back, does this PR (local project) still build? Might be one less thing to hunt down when we follow up with the package managers

File: Demo/Application/Features/Amex/BraintreeDemoAmexViewController.m [hunk 5]
```diff
@@ -1,6 +1,7 @@
 #import "BraintreeDemoAmexViewController.h"
 @import BraintreeAmericanExpress;
 @import BraintreeCard;
+@import BraintreeCore;
```
[the remark this answers is not in the document]
Comment (jaxdesmarais) [reply Demo/Application/Features/Amex/BraintreeDemoAmexViewController.m:4]: Seems like there are some imports that were handled when importing other modules in Obj-C for us. Removing this gives us the error:
```
Receiver 'BTAPIClient' for class message is a forward declaration
```
On the following line in this file (and similarly in other files):
```
_apiClient = [[BTAPIClient alloc] initWithAuthorization:authorization];
```

File: IntegrationTests/Braintree-API-Integration-Specs/BTAPIClient_IntegrationTests.m (hunk 2 above)
Comment (jaxdesmarais) [reply IntegrationTests/Braintree-API-Integration-Specs/BTAPIClient_IntegrationTests.m:2]: I believe the `@import` should be okay for integration tests as the tests are passing currently. But I can update to `#import <BraintreeCore/BraintreeCore-Swift.h>` if that's our preferred way to import for these tests.

File: Sources/BraintreeApplePay/BTApplePayCardNonce.m (hunk 3 above)
Comment (jaxdesmarais) [reply Sources/BraintreeApplePay/BTApplePayCardNonce.m:1]: Similarly, seems like this was handled in other files that are now removed when the Obj-C files were deleted from core, so removing this gives us unknown type errors.

File: Sources/BraintreeCore/Public/BraintreeCore/Braintree.h (hunk 4 above)
Comment (jaxdesmarais) [reply Sources/BraintreeCore/Public/BraintreeCore/Braintree.h:8]: Yep, based on our findings yesterday we will. Adding back in this PR to keep the size of the follow up PR smaller. Also updating a number of the imports here vs in the follow up PR.

**Review by scannillo — APPROVED** [review]
Pulled this locally and it builds OK for me. Thanks for the thorough responses to my questions, too! Super helpful.

**Pull request merged**